src/test/java/io/lettuce/core/cluster/models/partitions/RedisClusterNodeUnitTests.java

```diff
@@ -16,6 +18,55 @@
  */
 class RedisClusterNodeUnitTests {
 
+    @Test
+    void shouldCreateNodeWithEmptySlots() {
+
+        BitSet slots = new BitSet();
+        RedisClusterNode node = new RedisClusterNode(RedisURI.create("localhost", 6379), "1", true, null, 0, 0, 0, slots,
+                Collections.emptySet());
+        assertThat(node.getSlots()).isEmpty();
+        assertThat(node.getSlots()).isNotNull();
+    }
+
+    @Test
+    void shouldCreateNodeWithNonEmptySlots() {
+
+        BitSet slots = new BitSet();
+        slots.set(1);
+        slots.set(2);
+        RedisClusterNode node = new RedisClusterNode(RedisURI.create("localhost", 6379), "1", true, null, 0, 0, 0, slots,
+                Collections.emptySet());
+
+        assertThat(node.getSlots()).containsExactly(1, 2);
+    }
+
+    @Test
+    void shouldCopyNodeWithEmptySlots() {
+
+        BitSet slots = new BitSet();
+        RedisClusterNode originalNode = new RedisClusterNode(RedisURI.create("localhost", 6379), "1", true, null, 0, 0, 0,
+                slots, Collections.emptySet());
+
+        RedisClusterNode copiedNode = new RedisClusterNode(originalNode);
+
+        assertThat(copiedNode.getSlots()).isEmpty();
+        assertThat(copiedNode.getSlots()).isNotNull();
+    }
+
+    @Test
+    void shouldCopyNodeWithNonEmptySlots() {
+
+        BitSet slots = new BitSet();
+        slots.set(1);
+        slots.set(2);
+        RedisClusterNode originalNode = new RedisClusterNode(RedisURI.create("localhost", 6379), "1", true, null, 0, 0, 0,
+                slots, Collections.emptySet());
+
+        RedisClusterNode copiedNode = new RedisClusterNode(originalNode);
+
+        assertThat(copiedNode.getSlots()).containsExactly(1, 2);
+    }
+
```

Comment:
I think the most important test is the one described in #2341 and it is missing: 
- create one RedisClusterNode instance from the constructor on line 102-103 (parsed nodes use this constructor)
- create one RedisClusterNode instance from the constructor on line 112-125 (cloned nodes use this constructor)
- check if `RedisClusterNode::hasSameSlotsAs()` returns that they are the same

If you see the last mention of an issue in the original issue you'll see that there is also a 3rd constructor (the one on line 78-79) that calls the `setSlotBits()` method and it would also result in an null `slots` field if the array is empty. If we had a test for that we would have caught that this condition is not handled right now.

I think a good design would address this and make sure there is no way we set null to the slots field. What do you think?

Reply:
Thank you. I forgot to add the necessary test. I've added it. Could you please review again?